util/hashdb/src/lib.rs

```diff
@@ -17,62 +17,77 @@
 //! Database of byte-slices keyed to their Keccak hash.
 extern crate elastic_array;
 extern crate ethereum_types;
+extern crate heapsize;
+extern crate tiny_keccak;
 
-use std::collections::HashMap;
 use elastic_array::ElasticArray128;
 use ethereum_types::H256;
+use heapsize::HeapSizeOf;
+use std::collections::HashMap;
+use std::{fmt::Debug, hash::Hash};
+use tiny_keccak::Keccak;
+
+pub trait Hasher: Sync + Send {
+	type Out: AsRef<[u8]> + Debug + PartialEq + Eq + Clone + Copy + Hash + Send + Sync /* REVIEW: how do I get around this? --> */ + HeapSizeOf;
```

Comment:
Most of these trait requirements are specific to particular functions using `Hasher` and imo only those functions should require a template type to implement them.